lib/ace/transport_app.rb

```diff
@@ -251,8 +254,14 @@ def self.trusted_facts(certname)
 
         parameters = body['parameters'] || {}
 
-        # Since this will only be on one node we can just return the first result
+        # Unset gem specifications so that LOAD_PATH is loaded first, then
+        # append gem paths to the LOAD_PATH so they are still loaded. This is
+        # limited to the task run environment itself.
+        ENV['BOLT_ORIG_GEM_PATH'] = ""
+        ENV['BOLT_ORIG_GEM_HOME'] = ""
+        ENV['BOLT_ORIG_RUBYLIB'] = $LOAD_PATH.clone.concat(Gem.path).join(":")
```

Comment:
Is it safe to set these in threads? Also, why not set them in in ace-vanagon when the service starts?

Reply:
Oo that's a good idea - I'm not sure if `$LOAD_PATH` gets modified, but at least GEM_PATH and GEM_HOME can definitely be in the service file.